tests/acceptance/helpers/webdavHelper.js

```diff
@@ -34,11 +37,25 @@ exports.download = function(userId, file) {
  * @param {string} userId
  * @param {string} file
  */
-exports.delete = function(userId, file) {
+exports.delete = async function(userId, file) {
   const davPath = exports.createDavPath(userId, file)
+
+  /**
+   * makes sure delete operations are carried out maximum once a second to avoid trashbin issues
+   * see https://github.com/owncloud/core/issues/23151
+   */
+  const timeSinceLastDelete = Date.now() - timeOfLastDeleteOperation
+  if (timeSinceLastDelete < 1001) {
+    await client.pause(1001 - timeSinceLastDelete)
+  }
+  timeOfLastDeleteOperation = Date.now()
```

Comment:
I guess this is not essential to do here, because `timeOfLastDeleteOperation` will be set after `.delete(davPath, userId)` happens. But it does not hurt.

Reply:
I've deleted that extra line